lib/datix.ex

```diff
@@ -372,61 +376,70 @@ defmodule Datix do
     {:error, %FormatStringError{reason: {:invalid_modifier, modifier_to_string(modifier)}}}
   end
 
-  defp parse_integer(str, padding, width, int \\ nil)
+  defp parse_integer(str, padding, width, exact_width?, int \\ nil)
 
-  defp parse_integer("-" <> int_str, padding, width, nil) do
-    with {:ok, int, rest} <- parse_pos_integer(int_str, padding, width, 0) do
+  defp parse_integer("-" <> int_str, padding, width, exact_width?, nil) do
+    with {:ok, int, rest} <- parse_pos_integer(int_str, padding, width, exact_width?, nil) do
       {:ok, int * -1, rest}
     end
   end
 
-  defp parse_integer(int_str, padding, width, nil) do
-    parse_pos_integer(int_str, padding, width, nil)
+  defp parse_integer(int_str, padding, width, exact_width?, nil) do
+    parse_pos_integer(int_str, padding, width, exact_width?, nil)
   end
 
-  defp parse_pos_integer(str, int \\ nil)
-
-  defp parse_pos_integer(<<digit, rest::binary>>, int) when digit in ?0..?9 do
-    parse_pos_integer(rest, (int || 0) * 10 + (digit - ?0))
+  defp parse_pos_integer(str) do
+    case Integer.parse(str) do
+      {int, rest} -> {:ok, int, rest}
+      :error -> {:error, %ParseError{reason: :invalid_integer}}
+    end
```

Comment:
This is a small refactor to use a built-in function for parsing normal integers.